### PR TITLE
feat: add server dockerfile

### PR DESCRIPTION
**Pull request by milhamsuryapratama**

### CODE REVIEW

File: apps/server/project.json [hunk 1]
```diff
@@ -17,6 +17,27 @@
         }
       ]
     },
+    "build:linux": {
```
Comment (mnindrazaka) [inline apps/server/project.json:20]: `build-linux`

File: apps/server/project.json [hunk 2]
```diff
@@ -17,6 +17,27 @@
...
+      "executor": "@nx-go/nx-go:build",
+      "options": {
+        "outputPath": "apps/server/dist/",
+        "main": "apps/server/main.go"
+      }
+    },
+    "build:image": {
```
Comment (mnindrazaka) [inline apps/server/project.json:27]: `image-build`

File: apps/server/project.json [hunk 3]
```diff
@@ -17,6 +17,27 @@
...
+      "executor": "nx:run-commands",
+      "options": {
+        "cwd": "apps/server",
+        "command": "docker build --platform linux/amd64 -t gatherloop/starter-monorepo-server ."
+      }
+    },
+    "push:image": {
```
Comment (mnindrazaka) [inline apps/server/project.json:34]: `image-push`

Comment (milhamsuryapratama) [reply apps/server/project.json]: done paman @mnindrazaka

File: apps/server/project.json (hunk 2 above)
Comment (milhamsuryapratama) [reply apps/server/project.json]: done paman @mnindrazaka

File: apps/server/project.json (hunk 1 above)
Comment (milhamsuryapratama) [reply apps/server/project.json]: done paman @mnindrazaka

**Review by mnindrazaka — APPROVED** [review]

**Pull request merged**